### PR TITLE
MOD-7302: Fix numeric range syntax parser/v2

### PR DESCRIPTION
**Pull request by nafraf**
Backport to dialect 2 the fixes to numeric range syntax:

1. Support parameters with MINUS sign:
```sh
127.0.0.1:6379> FT.explaincli idx '@n:[-$p 500]' params 2 p 10 dialect 2
1) NUMERIC {-10.000000 <= @n <= 500.000000}
```
 2. Support exponents with PLUS sign:
 ```sh
127.0.0.1:6379> FT.explaincli idx '@n:[1 2e+3]' dialect 2
1) NUMERIC {1.000000 <= @n <= 2000.000000}
```
3. Support Inf with case insensitivity:
```sh
127.0.0.1:6379> FT.explaincli idx '@n:[0 INF]' dialect 2
1) NUMERIC {0.000000 <= @n <= inf}
```

**Which issues this PR fixes**
1. MOD-7302

**Main objects this PR modified**
1. ...
2. ...

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: src/query_parser/v2/lexer.rl [hunk 1]
```diff
@@ -29,9 +29,9 @@ void RSQuery_ParseFree_v2(void *p, void (*freeProc)(void *));
 
 machine query;
 
-inf = ['+\-']? 'inf' $ 4;
+inf = ('+'|'-')? 'inf'i $ 4;
 size = digit+ $ 2;
-number = '-'? digit+('.' digit+)? (('E'|'e') '-'? digit+)? $ 3;
+number = ('+'|'-')? (digit+('.' digit*)? | ('.' digit+)) (('E'|'e') ('+'|'-')? digit+)? $ 3;
```
Comment (oshadmi) [inline src/query_parser/v2/lexer.rl:34]: As discussed, this is a breaking change - need to do it in a different PR when we can break
```suggestion
number = ('+'|'-')? digit+('.' digit+)? (('E'|'e') ('+'|'-')? digit+)? $ 3;
```

Comment (nafraf) [reply src/query_parser/v2/lexer.rl]: OK. Change reverted

File: src/query_parser/v2/lexer.rl [hunk 2]
```diff
@@ -29,9 +29,9 @@ void RSQuery_ParseFree_v2(void *p, void (*freeProc)(void *));
 
 machine query;
 
-inf = ['+\-']? 'inf' $ 4;
+inf = ('+'|'-')? 'inf'i $ 4;
```
Comment (raz-mon) [inline src/query_parser/v2/lexer.rl:32]: Consider leaving this as it is, it's quite nice :)

File: src/query_parser/v2/lexer.rl [hunk 3]
```diff
@@ -29,9 +29,9 @@ void RSQuery_ParseFree_v2(void *p, void (*freeProc)(void *));
...
 size = digit+ $ 2;
-number = '-'? digit+('.' digit+)? (('E'|'e') '-'? digit+)? $ 3;
+number = ('+'|'-')? digit+('.' digit+)? (('E'|'e') ('+'|'-')? digit+)? $ 3;
```
Comment (raz-mon) [inline src/query_parser/v2/lexer.rl:34]: Same here, can use `[+\-]`.
Doesn't matter at all but I guess it's nice to use the specific syntax of the lexer.

File: src/query_parser/v2/parser.y [hunk 4]
```diff
@@ -1174,7 +1182,14 @@ exclusive_param_num(A) ::= LP num(B). {
 
 exclusive_param_num(A) ::= LP ATTRIBUTE(B). {
     A = B;
+    A.sign = 1;
```
Comment (raz-mon) [inline src/query_parser/v2/parser.y:1185]: Can also set the default value of `A.sign` to 1. I think it's not initialized now

File: src/query_parser/v2/parser.y [hunk 5]
```diff
@@ -1156,6 +1156,14 @@ param_size(A) ::= ATTRIBUTE(B). {
 
 param_num(A) ::= ATTRIBUTE(B). {
     A = B;
+    A.sign = 1; // default
```
Comment (raz-mon) [inline src/query_parser/v2/parser.y:1159]: Can be set in the `QueryToken` initialization.

File: tests/pytests/test_dialect.py [hunk 6]
```diff
@@ -139,24 +139,11 @@ def test_v1_vs_v2_vs_v5(env):
...
-      '      +e(expanded)',
-      '    }',
-      '    UNION {',
-      '      3',
-      '      +3(expanded)',
-      '    }',
-      '  }',
+      'UNION {',
+      '  1.2e+3',
+      '  +1.2e+3(expanded)',
```
Comment (raz-mon) [inline tests/pytests/test_dialect.py:144]: This will be fixed in a separate PR?

File: tests/pytests/test_dialect.py [hunk 7]
```diff
@@ -199,8 +186,8 @@ def test_v1_vs_v2_vs_v5(env):
       '      +e(expanded)',
       '    }',
       '    UNION {',
-      '      3',
-      '      +3(expanded)',
+      '      +3',
+      '      ++3(expanded)',
```
Comment (raz-mon) [inline tests/pytests/test_dialect.py:190]: Not so nice...

**Review by raz-mon — COMMENTED** [review]
Great.
See few comments

File: src/query_parser/v2/lexer.rl (hunk 2 above)
Comment (raz-mon) [reply src/query_parser/v2/lexer.rl:32]: I mean the `['+\-']` only

**Review by oshadmi — APPROVED** [review]

File: src/query_parser/v2/lexer.rl (hunk 2 above)
Comment (raz-mon) [reply src/query_parser/v2/lexer.rl:32]: BTW not sure that we need the quotes here.

File: tests/pytests/test_dialect.py (hunk 6 above)
Comment (nafraf) [reply tests/pytests/test_dialect.py:144]: Yes, it is part of MOD-7025

File: src/query_parser/v2/lexer.rl (hunk 2 above)
Comment (nafraf) [reply src/query_parser/v2/lexer.rl]: OK. Change done. Thanks.

File: src/query_parser/v2/parser.y (hunk 4 above)
Comment (nafraf) [reply src/query_parser/v2/parser.y]: OK. Done.

**Review by raz-mon — APPROVED** [review]
LGTM 👌
